tests/conftest.py

```diff
@@ -173,3 +173,7 @@ async def handler(request: web.Request) -> web.Response:
     finally:
         await site.stop()
         await runner.cleanup()
+    # Give the socket a chance to clean-up.
+    # This is specifically needed for PyPy3.10
+    # to avoid a ResourceWarning.
+    await asyncio.sleep(1)
```

Comment:
I think you can do `await site._server.wait_closed()` but it does require reaching into aiohttp internals. I don't know why `await site.stop()` does do this, but this issue seems to be the reasoning, https://github.com/aio-libs/aiohttp/issues/2746

Reply:
Thank you, I'll give that a try. Yeah, this seems like something that should get handled by the library.